From b9e9358cd70096f4a12b83708db0621d682d8cef Mon Sep 17 00:00:00 2001
From: Andrii Osipov <aosipov@anaconda.com>
Date: Wed, 21 Jan 2026 07:55:09 -0800
Subject: [PATCH] Add missing absl/strings/str_cat.h includes

When using system abseil instead of vendored abseil, some transitive
includes are not available. This adds explicit includes for str_cat.h
in tensorflow-text source files that use absl::StrCat or absl::StrAppend.
---
 tensorflow_text/core/kernels/boise_offset_converter_kernel_template.h  | 1 +
 tensorflow_text/core/kernels/byte_splitter_kernel_template.h           | 1 +
 tensorflow_text/core/kernels/fast_bert_normalizer.h                    | 1 +
 tensorflow_text/core/kernels/fast_bert_normalizer_model_builder.cc     | 1 +
 tensorflow_text/core/kernels/fast_bert_normalizer_test.cc              | 1 +
 tensorflow_text/core/kernels/fast_wordpiece_tokenizer.cc               | 1 +
 tensorflow_text/core/kernels/fast_wordpiece_tokenizer_test.cc          | 1 +
 tensorflow_text/core/kernels/phrase_tokenizer_model_builder.cc         | 1 +
 tensorflow_text/core/kernels/round_robin_trimmer_kernel_template.h     | 1 +
 tensorflow_text/core/kernels/sentence_fragmenter_v2_kernel_template.h  | 1 +
 tensorflow_text/core/kernels/whitespace_tokenizer_kernel_template.h    | 1 +
 11 files changed, 11 insertions(+)

diff --git a/tensorflow_text/core/kernels/boise_offset_converter_kernel_template.h b/tensorflow_text/core/kernels/boise_offset_converter_kernel_template.h
index f49f059..8143976 100644
--- a/tensorflow_text/core/kernels/boise_offset_converter_kernel_template.h
+++ b/tensorflow_text/core/kernels/boise_offset_converter_kernel_template.h
@@ -21,6 +21,7 @@
 #include <vector>
 
 #include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
 #include "tensorflow/core/platform/tstring.h"
 #include "tensorflow/lite/kernels/shim/op_kernel.h"
 #include "tensorflow/lite/kernels/shim/shape.h"
diff --git a/tensorflow_text/core/kernels/byte_splitter_kernel_template.h b/tensorflow_text/core/kernels/byte_splitter_kernel_template.h
index 77ab2b1..2472fee 100644
--- a/tensorflow_text/core/kernels/byte_splitter_kernel_template.h
+++ b/tensorflow_text/core/kernels/byte_splitter_kernel_template.h
@@ -19,6 +19,7 @@
 #include <vector>
 
 #include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
 #include "tensorflow/core/platform/tstring.h"
 #include "tensorflow/lite/kernels/shim/op_kernel.h"
 #include "tensorflow/lite/kernels/shim/shape.h"
diff --git a/tensorflow_text/core/kernels/fast_bert_normalizer.h b/tensorflow_text/core/kernels/fast_bert_normalizer.h
index efd5102..55888d3 100644
--- a/tensorflow_text/core/kernels/fast_bert_normalizer.h
+++ b/tensorflow_text/core/kernels/fast_bert_normalizer.h
@@ -18,6 +18,7 @@
 #include <cstdint>
 #include <vector>
 
+#include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 #include "icu4c/source/common/unicode/utf8.h"
 #include "tensorflow/lite/kernels/shim/status_macros.h"
diff --git a/tensorflow_text/core/kernels/fast_bert_normalizer_model_builder.cc b/tensorflow_text/core/kernels/fast_bert_normalizer_model_builder.cc
index 808d2a0..c0dbcac 100644
--- a/tensorflow_text/core/kernels/fast_bert_normalizer_model_builder.cc
+++ b/tensorflow_text/core/kernels/fast_bert_normalizer_model_builder.cc
@@ -23,6 +23,7 @@
 #include "absl/memory/memory.h"
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 #include "icu4c/source/common/unicode/errorcode.h"
 #include "icu4c/source/common/unicode/normalizer2.h"
diff --git a/tensorflow_text/core/kernels/fast_bert_normalizer_test.cc b/tensorflow_text/core/kernels/fast_bert_normalizer_test.cc
index 84fae67..9a55dc2 100644
--- a/tensorflow_text/core/kernels/fast_bert_normalizer_test.cc
+++ b/tensorflow_text/core/kernels/fast_bert_normalizer_test.cc
@@ -18,6 +18,7 @@
 
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
+#include "absl/strings/str_cat.h"
 #include "absl/strings/str_join.h"
 #include "tensorflow_text/core/kernels/fast_bert_normalizer_model_builder.h"
 
diff --git a/tensorflow_text/core/kernels/fast_wordpiece_tokenizer.cc b/tensorflow_text/core/kernels/fast_wordpiece_tokenizer.cc
index 88feee1..bde55dd 100644
--- a/tensorflow_text/core/kernels/fast_wordpiece_tokenizer.cc
+++ b/tensorflow_text/core/kernels/fast_wordpiece_tokenizer.cc
@@ -20,6 +20,7 @@
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
 #include "absl/strings/match.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/str_join.h"
 #include "absl/strings/string_view.h"
 #include "icu4c/source/common/unicode/uchar.h"
diff --git a/tensorflow_text/core/kernels/fast_wordpiece_tokenizer_test.cc b/tensorflow_text/core/kernels/fast_wordpiece_tokenizer_test.cc
index 118df70..aa7c6b1 100644
--- a/tensorflow_text/core/kernels/fast_wordpiece_tokenizer_test.cc
+++ b/tensorflow_text/core/kernels/fast_wordpiece_tokenizer_test.cc
@@ -17,6 +17,7 @@
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 #include "absl/flags/flag.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/str_join.h"
 #include "icu4c/source/common/unicode/uchar.h"
 #include "tensorflow/core/platform/env.h"
diff --git a/tensorflow_text/core/kernels/phrase_tokenizer_model_builder.cc b/tensorflow_text/core/kernels/phrase_tokenizer_model_builder.cc
index 268aeb3..f7d1751 100644
--- a/tensorflow_text/core/kernels/phrase_tokenizer_model_builder.cc
+++ b/tensorflow_text/core/kernels/phrase_tokenizer_model_builder.cc
@@ -23,6 +23,7 @@
 #include "absl/container/flat_hash_map.h"
 #include "absl/container/flat_hash_set.h"
 #include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 #include "tensorflow/lite/kernels/shim/status_macros.h"
 #include "tensorflow_text/core/kernels/phrase_tokenizer_model_generated.h"
diff --git a/tensorflow_text/core/kernels/round_robin_trimmer_kernel_template.h b/tensorflow_text/core/kernels/round_robin_trimmer_kernel_template.h
index 51f17da..d96016b 100644
--- a/tensorflow_text/core/kernels/round_robin_trimmer_kernel_template.h
+++ b/tensorflow_text/core/kernels/round_robin_trimmer_kernel_template.h
@@ -20,6 +20,7 @@
 #include <vector>
 
 #include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
 #include "absl/types/span.h"
 #include "tensorflow/lite/kernels/shim/op_kernel.h"
 #include "tensorflow/lite/kernels/shim/shape.h"
diff --git a/tensorflow_text/core/kernels/sentence_fragmenter_v2_kernel_template.h b/tensorflow_text/core/kernels/sentence_fragmenter_v2_kernel_template.h
index ea7be58..7b5c0a2 100644
--- a/tensorflow_text/core/kernels/sentence_fragmenter_v2_kernel_template.h
+++ b/tensorflow_text/core/kernels/sentence_fragmenter_v2_kernel_template.h
@@ -20,6 +20,7 @@
 #include <vector>
 
 #include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
 #include "tensorflow/lite/kernels/shim/op_kernel.h"
 #include "tensorflow/lite/kernels/shim/shape.h"
 #include "tensorflow/lite/kernels/shim/status_macros.h"
diff --git a/tensorflow_text/core/kernels/whitespace_tokenizer_kernel_template.h b/tensorflow_text/core/kernels/whitespace_tokenizer_kernel_template.h
index e05a914..785e7d9 100644
--- a/tensorflow_text/core/kernels/whitespace_tokenizer_kernel_template.h
+++ b/tensorflow_text/core/kernels/whitespace_tokenizer_kernel_template.h
@@ -20,6 +20,7 @@
 
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
+#include "absl/strings/str_cat.h"
 #include "tensorflow/core/framework/tensor.h"
 #include "tensorflow/lite/kernels/shim/op_kernel.h"
 #include "tensorflow/lite/kernels/shim/shape.h"
